src/avs/lagrangeZK/lagrangeZK.go

```diff
@@ -0,0 +1,111 @@
+package lagrangezk
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/config"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/eigenlayer"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/etherfi"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/gnosis"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/utils"
+)
+
+// API handle for all core lagrangeZK functionality
+type API struct {
+	Client *ethclient.Client
+
+	ServiceManager            *ZKMRServiceManager
```

Comment:
Are you just including this to keep the APIs consistent? Doesn't seem like we ever interact with the service manager directly just use the address as a param for the `OperatorAVSRegistrationDigestHash`

Reply:
yeah. Doesn't need to be there but I think long term its likely there are some calls we would need to support such as changingQuorums and such. Doesn't matter too much either way